app/Http/Controllers/FileController.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests\CSVFormRequest;
+use App\Jobs\ReadFileJob;
+use Illuminate\Support\Facades\Storage;
+
+class FileController extends Controller
+{
+    /**
+     * Function uploadFile
+     *
+     * @return JsonResponse
+     */
+    public function uploadFile(CSVFormRequest $request)
+    {
+        $path = $request->file('csv')->getRealPath();
```

Comment:
follow chỗ ni sẽ như ri. 
- Upload file => lưu file vào 1 chỗ nào đó. https://laravel.com/docs/7.x/filesystem#the-local-driver 
- Sau đó dispatch Job để đọc file và xử lý file. 
- Sau khi xử lý xong thì xóa cái file đó đi cho nhẹ.

Reply:
chị thuỷ bảo ko cần trong case này https://neolab.wc.calling.fun/group/mi2-new-backend-0322?msg=6zTxc26Cmc7CnHDX9

Reply:
C nói không cần lưu khi trong yêu cầu không có thôi. 
Nhưng mới đọc lại msg của Mỹ thì có upload file lên server đó chú =))

Reply:
good job mentor